packages/transaction-controller/src/TransactionController.ts

```diff
@@ -610,9 +619,94 @@ export class TransactionController extends BaseControllerV1<
       this.onBootCleanup();
     });
 
+    this.onNetworkStateChange = onNetworkStateChange;
+    this.isMultichainEnabled = isMultichainEnabled;
+    this.incomingTransactions = incomingTransactions;
+    this.getNetworkClientRegistry = getNetworkClientRegistry;
+
     this.onBootCleanup();
   }
 
+  /**
+   * Initializes `registry`, `nonceTracker`, `#multichainTrackingHelper`,
+   * `incomingTransactionHelper`, `pendingTransactionTracker`, and passes a
+   * callback to `onNetworkStateChange`.
+   *
+   * This relies on the provider and block tracker being defined which may not
+   * have been the case at the time the Transaction Controller was instantiated.
+   *
+   * @param passedProvider - Reference to the provider proxy object
+   * @param passedBlockTracker - Reference to the block tracker proxy object
+   */
+  delayedInit(passedProvider: Provider, passedBlockTracker: BlockTracker) {
+    // @ts-expect-error the type in eth-method-registry is inappropriate and should be changed
+    this.registry = new MethodRegistry({ provider: passedProvider });
+
+    this.nonceTracker = this.#createNonceTracker({
```

Comment:
We still have all the original initialisations so I worry that makes the solution even more complex since we don't know if we have a functional helper (nonce tracker, multi-chain tracker etc) or a functioning one?

Plus having these helpers initialised only once allows us to maintain stability by being confident we are dealing with single instances with one set of event listeners for example, and not being as concerned about deconstruction.